src/streamsync/core.py

```diff
@@ -722,6 +739,93 @@ def call_frontend_function(self, module_key: str, function_name: str, args: List
         })
 
 
+class EventHandlerRegistry:
+    """
+    Maps functions registered as event handlers from the user app's core
+    and external modules, providing an access mechanism to these maps.
+    """
+
+    HandlerArgsSequence = Sequence[str]
+    HandlerMetaEntry = Dict[str, Union[str, HandlerArgsSequence]]
+    HandlerEntry = Dict[str, Union[Callable, HandlerMetaEntry]]
+
+    def __init__(self):
+        self.handler_map: Dict[str, 'EventHandlerRegistry.HandlerEntry'] = {}
+
+    def __iter__(self):
+        return iter(self.handler_map.keys())
+
+    def register_handler(self, handler: Callable):
+        module_name = handler.__module__
+
+        # Prepare "access name"
+        # (i.e. the key that frontend uses to retrieve handler)
+        if module_name == "streamsyncuserapp":
+            # Use the handler's __qualname__ directly
+            # for functions from main.py in user's app
+            access_name = handler.__qualname__
+        else:
+            # For external handlers, separate the module name
+            # and handler __qualname__by a dot
+            access_name = f"{module_name}.{handler.__qualname__}"
+
+        entry: EventHandlerRegistry.HandlerEntry = \
+            {
+                "callable": handler,
+                "meta": {
+                    "name": access_name,
+                    "args": inspect.getfullargspec(handler).args
+                }
+            }
+
+        self.handler_map[access_name] = entry
+
+    def register_module(self, module: ModuleType):
+        if isinstance(module, ModuleType):
+            all_fn_names = (x[0] for x in inspect.getmembers(
+                module, inspect.isfunction))
+            exposed_fn_names = list(
+                filter(lambda x: not x.startswith("_"), all_fn_names))
+
+            for fn_name in exposed_fn_names:
+                fn_callable = getattr(module, fn_name)
+                if not fn_callable:
+                    continue
+                self.register_handler(fn_callable)
+        else:
+            raise ValueError(
+                f"Attempted to register a non-module object: {module}"
+                )
+
+    def find_handler(self, handler_name: str) -> Optional[Callable]:
+        handler_entry: EventHandlerRegistry.HandlerEntry = \
+            self.handler_map.get(handler_name, {})
+        handler_callable = \
+            cast(Optional[Callable], handler_entry.get("callable"))
+        return handler_callable
+
+    def get_handler_meta(
+            self,
+            handler_name: str
+            ) -> "EventHandlerRegistry.HandlerMetaEntry":
+        if handler_name not in self.handler_map:
+            raise RuntimeError(f"Handler {handler_name} is not registered")
+        entry: EventHandlerRegistry.HandlerEntry = \
+            self.handler_map[handler_name]
+        if "meta" not in entry:
```

Comment:
that should not happen ? it set inside `register_handler`.

Reply:
I vaguely remembered adding this because mypy was upset 😄 but now, verifying it, seems that it is not the case. Good point, I'll fix this.